airbrakes/mock/display.py

```diff
@@ -0,0 +1,104 @@
+        """
+        Updates the display with real-time data.
+        :param end_sim: Whether the simulation ended or was interrupted.
+        """
+        try:
+            current_queue_size = self.airbrakes.imu._data_queue.qsize()
+        except NotImplementedError:  # Returns NotImplementedError on arm architecture (Raspberry Pi)
+            current_queue_size = "N/A"
+
+        # Prepare output
+        output = [
+            f"{Y}{'=' * 12} REAL TIME FLIGHT DATA {'=' * 12}{RESET}",
+            f"Time since sim start:      {G}{time.time() - self.start_time:<10.2f}{RESET} {R}s{RESET}",
+            f"State:                     {G}{self.airbrakes.state.name:<15}{RESET}",
+            f"Current speed:             {G}{self.airbrakes.data_processor.speed:<10.2f}{RESET} {R}m/s{RESET}",
+            f"Max speed so far:          {G}{self.airbrakes.data_processor.max_speed:<10.2f}{RESET} {R}m/s{RESET}",
+            f"Current height:            {G}{self.airbrakes.data_processor.current_altitude:<10.2f}{RESET} {R}m{RESET}",
+            f"Max height so far:         {G}{self.airbrakes.data_processor.max_altitude:<10.2f}{RESET} {R}m{RESET}",
+            f"Airbrakes extension:       {G}{self.airbrakes.current_extension.value}{RESET}",
+            f"IMU Data Queue Size:       {G}{current_queue_size}{RESET}",
+            f"{Y}{'=' * 13} REAL TIME CPU LOAD {'=' * 14}{RESET}",
+        ]
+
+        # Add CPU usage data with color coding
+        for name, process in self.processes.items():
+            cpu_usage = process.cpu_percent(interval=None)
+            if cpu_usage < 50:
+                cpu_color = G
+            elif cpu_usage < 75:
+                cpu_color = Y
+            else:
+                cpu_color = R
+            output.append(f"{name:<25}    {cpu_color}CPU Usage: {cpu_usage:>6.2f}% {RESET}")
+
+        # Print the output
+        print("\n".join(output))
+
+        # Move the cursor up for the next update, if the simulation hasn't ended:
+        if not end_sim:
+            print(self.MOVE_CURSOR_UP * len(output), end="", flush=True)
+
+        # Print the end of simulation message if the simulation has ended
+        if end_sim == self.NATURAL_END:
+            # Print the end of simulation header
+            print(f"{Fore.RED}{'=' * 14} END OF SIMULATION {'=' * 14}{Style.RESET_ALL}")
+        elif end_sim == self.INTERRUPTED_END:
+            print(f"{Fore.RED}{'=' * 12} INTERRUPTED SIMULATION {'=' * 13}{Style.RESET_ALL}")
+
+    def prepare_process_dict(self) -> dict[str, psutil.Process]:
+        """
+        Prepares a dictionary of processes to monitor CPU usage for.
+        """
+        all_processes = {}
+        imu_process = self.airbrakes.imu._data_fetch_process
+        log_process = self.airbrakes.logger._log_process
+        current_process = multiprocessing.current_process()
+        for p in [imu_process, log_process, current_process]:
+            all_processes[p.name] = psutil.Process(p.pid)
```

Comment:
I'm gonna nit pick you here, method needs :return in docstring and maybe add a comment for this line for psutil

Reply:
what do you want the comment to tell? Also, I love nitpicks.

Reply:
probably what `psutil.Process(p.pid)` does -- someone unfamiliar to what is happening might get confused